### DISCUSSION TITLE
how to inline styles using bundle-text

### DISCUSSION THREAD

**Discussion by mrassili** [parcel-bundler/parcel#4708] [General]
Hello

Thanks for the awesome work everyone!

So I have installed parcel#v2 and I run this build command 
`parcel build src/{style.pcss,background.js,content-script.js} && cp src/icons/*.png src/manifest.json dist/`

which generates those files
<img width="379" alt="Screen Shot 2020-06-07 at 11 01 04 AM" src="https://user-images.githubusercontent.com/25288435/83965810-35e1fc80-a8ae-11ea-8c86-a43127891dcb.png">

But when I add
```js
import style from "bundle-text:./style.css";
```
in `src/content-script.js`

I get this error during build `Cannot resolve dependency 'bundle-text:./style.css'`

cc @mischnic @DeMoorJasper

**Comment by mischnic** [comment]
> ./style.css

> style.pcss

You need to decide which extension you want to use.

Also, you likely only want background.js and content-script.js as entrypoints.

**Reply by mrassili** [reply]
> > ./style.css
> 
> > style.pcss
> 
> You need to decide which extension you want to use.

isn't `pcss` file bundled into `css`?

> Also, you likely only want background.js and content-script.js as entrypoints.

Yes will experiment with that

**Comment by mrassili** [comment]
If I don't include `style.pcss` in the bundle command it's not bundled, what should be the way to include it?

**Comment by mischnic** [comment]
1. Until that PR is ~merged~ published to npm, `.pcss` doesn't work

2. An example:

index.js:
```js
import style from "bundle-text:./style.less";

console.log(style);
```
style.less:
```less
body {
	color: red;
}
```
`parcel build index.js` works as expected

**Comment by mrassili** [comment]
> Until that PR is merged, .pcss doesn't work

I'm actually using the latest changes it's merged now, no?

and I get a `style.css` output without problems

> 2. An example:
> 
> index.js:
> 
> ```js
> import style from "bundle-text:./style.less";
> 
> console.log(style);
> ```
> 
> style.less:
> 
> ```css-less
> body {
> 	color: red;
> }
> ```
> 
> `parcel build index.js` works as expected

I don't have an `index.js`, it's not possible to import the css text inside `content-script.js`?

**Reply by mischnic** [reply]
> I don't have an index.js, it's not possible to import the css text inside content-script.js?

`index.js` can be any javascript file.

**Reply by mrassili** [reply]
yes but why do I get `Cannot resolve dependency 'bundle-text:./style.css'` with both `./style.css` and `./style.pcss` even though `style.css` gets bundled

**Reply by mischnic** [reply]
Please provide a sample of your code so that I can reproduce that error

**Reply by mrassili** [reply]
okay, so I do `yarn add parcel#v2`

`package.json`
```
  "scripts": {
    "prebuild": "rm -rf dist",
    "build": "parcel build src/{style.pcss,background.js,content-script.js} && cp src/icons/*.png src/manifest.json dist/"
  },
```

`src/content-script.js`
```
import style from "bundle-text:./style.css";
// also tried import style from "bundle-text:./style.pcss";

const shadowHost = document.createElement("div");
document.body.insertAdjacentElement("beforebegin", shadowHost);
const shadowRoot = shadowHost.attachShadow({ mode: "open" });
shadowRoot.innerHTML = `<style>${style}</style>`
```

**Reply by mischnic** [reply]
> yarn add parcel#v2

`yarn add parcel@next`. Your command still installs Parcel 1

**Reply by mrassili** [reply]
I managed to get it to work with simple css file using `parcel@next`

In `background.js`
```
import "./style.css";
import style from "bundle-text:./style.css";

console.log({ style })
```

`src/style.css`
```
body {
  font-family: "Montserrat", sans-serif;
  color: #18191f;
}
```

but all I get in the console is an empty style
<img width="345" alt="Screen Shot 2020-06-07 at 12 05 07 PM" src="https://user-images.githubusercontent.com/25288435/83967206-52cefd80-a8b7-11ea-991e-6d1f65cf3c6b.png">

EDIT: The build command results in this
```
dist/background.js
dist/background.css
dist/content-script.js
```

**Reply by mischnic** [reply]
Do you actually need `import "./style.css";`? 

I've created an issue for that bug: https://github.com/parcel-bundler/parcel/issues/4709

**Reply by mrassili** [reply]
> Do you actually need `import "./style.css";`?
> 

This is what I get when I don't include that line
```
dist/background.js
dist/content-script.js
```

**Reply by mischnic** [reply]
What are you trying to achieve here?

If you want to inject the CSS contents into some html element yourself, then this is fine?

**Reply by mrassili** [reply]
okay so the optimal result I am trying to accomplish is:
I have a `style.pcss` file and `content-script.js` that injects the bundled `css` into the html during build. This is to be able to inline styles in a shadow root element since outside styles can't affect it.

I can't write css directly inside the html because I use TailwindCSS and only write classes

If you can think of a better approach please let me know

**Reply by mrassili** [reply]
okay I can actually get rid of the `import './style.css'` and the styles are injected into `<style>${style}</style>` without problems
